azure-pipelines.yml

```diff
@@ -54,17 +57,20 @@ jobs:
       pytest-xdist
       scikit-learn
       scipy
-      "h5py<3.4.0"
       tqdm
       threadpoolctl
     displayName: 'Install dependencies'
+  - script: >-
+      python -m pip install --only-binary=h5py
+      "h5py<3.4.0"
+    displayName: 'Install h5py for Python < 3.10'
+    condition: and(succeeded(), ne(variables['PYTHON_VERSION'], '3.10'))
```

Comment:
we could try removing this condition, though we'll need to unpin `h5py` to get the binary on newer Python, and not sure how that will work out..